src/lib/ngsi/Scope.cpp

```diff
@@ -308,6 +306,46 @@ int Scope::fill
     return -1;
   }
 
+  //pointVectorRelease(pointV); ??
+  pointV.clear();
+
+  // Check invalid combinations
+  if ((geometry.areaType == "line") && (georel.type == "coveredBy"))
+  {
+    /* It seems that MongoDB 3.2 doesn't support this kind of queries, we get this error:
+     *
+     *  { $err: "Can't canonicalize query: BadValue $within not supported with provided geometry:
+     *    { $geoWithin: { $geometry: { type: "LineString", coordinates: [ [ 5.0...", code: 17287 }
+     */
+
+    *errorStringP = "line geometry cannot be used with coveredBy georel";
+    return -1;
+  }
+
+  if ((geometry.areaType == "point") && (georel.type == "coveredBy"))
+  {
+    /* It seems that MongoDB 3.2 doesn't support this kind of queries, we get this error:
+     *
+     *  { $err: "Can't canonicalize query: BadValue $within not supported with provided geometry:
+     *    { $geoWithin: { $geometry: { type: "Point", coordinates: [ [ 5.0...", code: 17287 }
+     */
+
+    *errorStringP = "point geometry cannot be used with coveredBy georel";
+    return -1;
+  }
+
+  if ((geometry.areaType != "point") && (georel.type == "near"))
```

Comment:
I removed the checking, but at the end it seems that MongoDB is not supporting $near with geometries different from Point, so I'm introducing it again (although probably not exactly the same way...).

NTC